x/stakeibc/keeper/unbonding_records.go

```diff
@@ -421,105 +415,98 @@ func (k Keeper) SweepAllUnbondedTokensForHostZone(ctx sdk.Context, hostZone type
 	redemptionCallback := types.RedemptionCallback{
-		HostZoneId:              hostZone.ChainId,
+		HostZoneId:              chainId,
 		EpochUnbondingRecordIds: epochUnbondingRecordIds,
 	}
 	marshalledCallbackArgs, err := k.MarshalRedemptionCallbackArgs(ctx, redemptionCallback)
 	if err != nil {
-		k.Logger(ctx).Error(err.Error())
-		return false, sdkmath.ZeroInt()
+		return err
 	}
 
-	// Send the transfer ICA
+	// Send the bank send ICA
 	_, err = k.SubmitTxsDayEpoch(ctx, hostZone.ConnectionId, msgs, types.ICAAccountType_DELEGATION, ICACallbackID_Redemption, marshalledCallbackArgs)
 	if err != nil {
-		k.Logger(ctx).Error(fmt.Sprintf("Failed to SubmitTxs, transfer to redemption account on %s", hostZone.ChainId))
-		return false, sdkmath.ZeroInt()
+		return errorsmod.Wrapf(err, "unable to submit redemption ICA for %s", hostZone.ChainId)
 	}
 	k.Logger(ctx).Info(utils.LogWithHostZone(hostZone.ChainId, "ICA MsgSend Successfully Sent"))
 
 	// Update the host zone unbonding records to status IN_PROGRESS
 	err = k.RecordsKeeper.SetHostZoneUnbondings(ctx, hostZone.ChainId, epochUnbondingRecordIds, recordstypes.HostZoneUnbonding_EXIT_TRANSFER_IN_PROGRESS)
 	if err != nil {
-		k.Logger(ctx).Error(err.Error())
-		return false, sdkmath.ZeroInt()
+		return err
 	}
 
-	return true, totalAmtTransferToRedemptionAcct
+	EmitRedemptionSweepEvent(ctx, hostZone, totalSweepAmount)
+
+	return nil
 }
 
-// Sends all unbonded tokens to the redemption account
-// returns:
-//   - success indicator if all chains succeeded
-//   - list of successful chains
-//   - list of tokens swept
-//   - list of failed chains
-func (k Keeper) SweepAllUnbondedTokens(ctx sdk.Context) (success bool, successfulSweeps []string, sweepAmounts []sdkmath.Int, failedSweeps []string) {
-	// this function returns true if all chains succeeded, false otherwise
-	// it also returns a list of successful chains (arg 2), tokens swept (arg 3), and failed chains (arg 4)
+// Sends all unbonded tokens that have finished unbonding to the redemption account
+func (k Keeper) SweepUnbondedTokensAllHostZones(ctx sdk.Context) {
 	k.Logger(ctx).Info("Sweeping All Unbonded Tokens...")
 
-	success = true
-	successfulSweeps = []string{}
-	sweepAmounts = []sdkmath.Int{}
-	failedSweeps = []string{}
-	hostZones := k.GetAllActiveHostZone(ctx)
-
-	epochUnbondingRecords := k.RecordsKeeper.GetAllEpochUnbondingRecord(ctx)
-	for _, hostZone := range hostZones {
-		hostZoneSuccess, sweepAmount := k.SweepAllUnbondedTokensForHostZone(ctx, hostZone, epochUnbondingRecords)
-		if hostZoneSuccess {
-			successfulSweeps = append(successfulSweeps, hostZone.ChainId)
-			sweepAmounts = append(sweepAmounts, sweepAmount)
-		} else {
-			success = false
-			failedSweeps = append(failedSweeps, hostZone.ChainId)
+	for _, hostZone := range k.GetAllActiveHostZone(ctx) {
+		err := utils.ApplyFuncIfNoError(ctx, func(ctx sdk.Context) error {
+			return k.SweepUnbondedTokensForHostZone(ctx, hostZone)
```

Comment:
So now that an error will roll back the state, we could still keep track of which ones errored to return a "failedSweeps" array but  I see the function output signature is changed.  Do we not have to worry about failed ones because we assume they will be picked up in the next round?

Reply:
that array was only used for the unit tests and it's pretty bad practice to return many values like this. So now that the tests are refactored, there's no longer a need for these